presto-main/src/main/java/io/prestosql/operator/window/LagFunction.java

```diff
@@ -33,6 +33,13 @@
     private final int offsetChannel;
     private final int defaultChannel;
     private final boolean ignoreNulls;
+    private long lastNonNull;
+
+    @Override
+    public void reset()
+    {
+        lastNonNull = 0;
+    }
```

Comment:
move below ctor

why FirstValueFunction uses -1 as "not set" and this uses 0?
i think 0 is not a good value for unset (to be verified)